apps/concierge_site/lib/helpers/mail_helper.ex

```diff
@@ -47,4 +47,14 @@ defmodule ConciergeSite.Helpers.MailHelper do
   def reset_password_url(reset_token) do
     Helpers.password_reset_url(ConciergeSite.Endpoint, :edit, reset_token)
   end
+
+  @spec track_open_url(%{alert_id: String.t(), notification_id: String.t()}) :: iodata
```

Comment:
The return type of URL helpers is `String.t()`, rather than the more general `iodata`.

Reply:
🤔 got it, is there a reason that dialyzer doesn't catch this?

Reply:
By default it's not an error if an explicit type is a superset of the success type; you need to enable the optional `underspecs` flag, which we haven't on this project. Trying that out might be a good idea, if it doesn't produce many errors on the existing code.

Reply:
cool, good to know